crates/core/src/workflows/hold_link.rs

```diff
@@ -80,7 +80,7 @@ pub async fn hold_link_workflow(
 
     log_debug!(context, "workflow/hold_link: added! {:?}", link);
 
-    //4. store link_add entry so we have all we need to respond to get links queries without any other network look-up
+    //4. store link_add entry so we have all we need to respond to get links queries without any other network look-up <== THIS!
```

Comment:
So as Nico mentioned this morning there is some logic for nodes to hold the LinkAdd entries for links on bases they hold!

I guess this kinda renders this whole PR redundant. It will actually result in more networks requests overall (assuming this logic for holding the LinkAdd entries actually works...)

Reply:
This is a possible future enhancement, but there are other problems getting this to work correctly, so for now we will merge this PR so that we can finish pagination which needs to be built on top of it.